community/www/profiles/profile.py

```diff
@@ -1,14 +1,16 @@
 import frappe
-from ...lms.doctype.lms_sketch.lms_sketch import get_recent_sketches
+from community.lms.models import Sketch
 
 def get_context(context):
     context.no_cache = 1
     context.username = frappe.form_dict["username"]
+    print(frappe.form_dict["username"])
     context.member = get_member(context.username)
+    print(context.member)
     if not context.member:
         context.template = "www/404.html"
     else:
-        context.sketches = list(filter(lambda x: x.owner == context.member.email, get_recent_sketches()))
+        context.sketches = list(filter(lambda x: x.owner == context.member.email, Sketch.get_recent_sketches()))
```

Comment:
The `get_recent_sketches` only returns the recent 100 sketches. So the number of sketches for that user could be very small or even none. The right way to do this is by adding a where clause in the db query.

Can you change the `get_recent_sketches` method to take an optional argument `owner` and add filters to `get_all` when `owner` is provided?